src/main/java/controllers/ConsoleInputController.java

```diff
@@ -0,0 +1,124 @@
+    private ProjectInputController projectInputController;
+    private String filePath = System.getProperty("user.dir") + "/savedProjects.json";
+
+    /**
+     * Constructor.
+     * @param view : takes in a View model, in this case a command line view.
+     */
+    public ConsoleInputController(CLIView view) {
+        this.consoleView = view;
+        loadProjectsData();
+        this.projectInputController = new ProjectInputController(this.consoleView, this.projectRepository);
+    }
+
+    /**
+     * Method that is called upon receiving commands from CLI.
+     * @param input : Input typed by user into CLI
+     */
+    @Override
+    public void onCommandReceived(String input) {
+        DukeLogger.logInfo(ConsoleInputController.class, "input:'" + input + "'");
+        Scanner inputReader = new Scanner(input);
+        String command = inputReader.next();
+
+        switch (command) {
+        case "bye":
+            consoleView.end();
+            break;
+        case "list":
+            ArrayList<Project> allProjects = projectRepository.getAll();
+            consoleView.viewAllProjects(allProjects);
+            break;
+        case "create":
+            // Creation of a new project with a given name and a number of numbers
+            boolean isProjectCreated = projectRepository.addToRepo(input);
+            if (!isProjectCreated) {
+                consoleView.consolePrint("Creation of Project failed. Please check parameters given!");
+            } else {
+                consoleView.consolePrint("Project created!");
+                saveProjectsData();
+            }
+            break;
+        case "manage":
+            if (inputReader.hasNext()) {
+                this.projectInputController.onCommandReceived(inputReader.next());
+            } else {
+                consoleView.consolePrint("Please enter a project number!");
+            }
+            break;
+        case "delete":
+            if (inputReader.hasNext()) {
+                int projectIndex = Integer.parseInt(inputReader.next());
+                boolean isProjectDeleted = this.projectRepository.deleteItem(projectIndex);
+                if (isProjectDeleted) {
+                    consoleView.consolePrint("Project " + projectIndex + " has been deleted");
+                } else {
+                    consoleView.consolePrint("Index out of bounds! Please check project index!");
+                }
+            } else {
+                consoleView.consolePrint("Please enter a project number to delete");
+            }
+            break;
+        case "help":
+            // TODO help page displaying all commands available
+            // Not implemented
+            consoleView.consolePrint("Not implemented");
+            break;
+        default:
+            consoleView.consolePrint("Invalid inputs. Please refer to User Guide or type help!");
+            break;
+        }
+    }
+
+
+    private void saveProjectsData() {
```

Comment:
A controller must not directly deal with the file system. Delegate this to a storage module.